sdl_android/src/main/java/com/smartdevicelink/streaming/RTPH264Packetizer.java

```diff
@@ -0,0 +1,488 @@
+	 */
+	@Override
+	public void start() throws IOException {
+		if (mThread != null) {
+			return;
+		}
+
+		mThread = new Thread(this);
+		mThread.start();
+	}
+
+	/**
+	 * Stops this packetizer.
+	 *
+	 * It is recommended that the video encoder is stopped prior to the packetizer.
+	 */
+	@Override
+	public void stop() {
+		if (mThread == null) {
+			return;
+		}
+
+		mThread.interrupt();
+		mThread = null;
+
+		mPaused = false;
+		mWaitForIDR = false;
+		mOutputQueue.clear();
+	}
+
+	/**
+	 * Pauses this packetizer.
+	 *
+	 * This pauses the packetizer but does not pause the video encoder.
+	 */
+	@Override
+	public void pause() {
+		mPaused = true;
+	}
+
+	/**
+	 * Resumes this packetizer.
+	 */
+	@Override
+	public void resume() {
+		mWaitForIDR = true;
+		mPaused = false;
+	}
+
+	/**
+	 * The thread routine.
+	 */
+	public void run() {
+		SdlConnection connection = _session.getSdlConnection();
+
+		while (mThread != null && !mThread.isInterrupted()) {
+			ByteBuffer frame;
+			try {
+				frame = mOutputQueue.take();
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				break;
+			}
+
+			while (frame.hasRemaining()) {
+				int len = frame.remaining() > bufferSize ? bufferSize : frame.remaining();
+
+				ProtocolMessage pm = new ProtocolMessage();
+				pm.setSessionID(_rpcSessionID);
+				pm.setSessionType(_serviceType);
+				pm.setFunctionID(0);
+				pm.setCorrID(0);
+				pm.setData(frame.array(), frame.arrayOffset() + frame.position(), len);
+				pm.setPayloadProtected(mServiceProtected);
+
+				_streamListener.sendStreamPacket(pm);
+
+				frame.position(frame.position() + len);
+			}
+		}
+
+		// XXX: This is added to sync with StreamPacketizer. Actually it shouldn't be here since
+		// it's confusing that a packetizer takes care of End Service request.
+		if (connection != null) {
+			connection.endService(_serviceType, _rpcSessionID);
+		}
+	}
+
+	/**
+	 * Called by the encoder.
+	 *
+	 * @see com.smartdevicelink.encoder.IEncoderListener#onEncoderOutput(Format, byte[], long)
+	 */
+	@Override
+	public void onEncoderOutput(IEncoderListener.Format format, byte[] data, long ptsInUs) {
+		if (data == null || format != Format.H264_BYTE_STREAM) {
+			return;
+		}
+
+		mNALUnitReader.init(data);
+		onEncoderOutput(mNALUnitReader, ptsInUs);
+	}
+
+	/**
+	 * Called by the encoder.
+	 *
+	 * @see com.smartdevicelink.encoder.IEncoderListener#onEncoderOutput(Format, byte[], int, int, long)
+	 */
+	@Override
+	public void onEncoderOutput(IEncoderListener.Format format, byte[] data,
+	                            int offset, int length, long ptsInUs) throws ArrayIndexOutOfBoundsException {
+		if (data == null || format != Format.H264_BYTE_STREAM) {
+			return;
+		}
+
+		mNALUnitReader.init(data, offset, length);
+		onEncoderOutput(mNALUnitReader, ptsInUs);
+	}
+
+	private void onEncoderOutput(NALUnitReader nalUnitReader, long ptsInUs) {
+		if (mPaused) {
+			return;
+		}
+
+		ByteBuffer nalu;
```

Comment:
This should have a little bit more descriptive of a name, maybe `nalUnit`. Changes should be applied to the `outputRTPFrames` signature too.

Reply:
Sure, I've renamed all `nalu` to `nalUnit` with commit https://github.com/smartdevicelink/sdl_android/pull/592/commits/eb79ad607e8a1ffe55138db37e07517f38e6e7d9.